### DISCUSSION TITLE
If using simAddVehicle() to spawn vehicles and then calling simGetVehiclePose()

### DISCUSSION THREAD

**Discussion by jconde42** [microsoft/AirSim#3844] [General]
I don't know if this is intended behavior but it doesn't seem to be documented. When spawning in vehicles with simAddVehicle() to specific points ( not the origin ) any subsequent call to simGetVehiclePose().position will be relative the the spawn location and not to the origin. Hopefully this helps others out there struggling.

**Comment by alonfaraj** [comment]
hi @jconde42,
I don't think it's a desired behavior. 
Also, it's seems like the same behavior reproduce when spawning from `settings.json` in different position than (0,0,0).
The reason is when you call `simGetVehiclePose()`, it's actually calling to `PawnSimApi::getPose` and the inner function calls to `NedTransform::toLocalNed`, but in this case you want to call `NedTransform::toGlobalNed`.

Maybe it worth to open an issue or you can make a PR so the maintainers confirm this is not the expected result.

**Comment by zimmy87** [comment]
See my comment [here](https://github.com/microsoft/AirSim/issues/4261#issuecomment-1009877677). I will discuss with my team whether we want to change this behavior, but it appears this may be by design. You can work around this by using the `simGetObjectPose()` API.
